pkg/controller.v1/tensorflow/controller.go

```diff
@@ -248,7 +248,12 @@ func (tc *TFController) processNextWorkItem() bool {
 	if err != nil {
 		if err == errNotExists {
 			logger.Infof("TFJob has been deleted: %v", key)
-			tfJobsDeletedCount.WithLabelValues(tfJob.Namespace).Inc()
+			namespace, _, keyerr := cache.SplitMetaNamespaceKey(key)
+			if keyerr == nil && len(namespace) != 0 {
+				tfJobsDeletedCount.WithLabelValues(namespace).Inc()
+			} else {
+				logger.Errorf("Invalid TFJob key %s: Namespace is missing %v", key, keyerr)
```

Comment:
Any case we've seen a job missing namespace?

Reply:
@johnugeorge

Reply:
I haven't. But, I have seen this condition used in other projects as well. Not sure, if it is valid now.

Reply:
Got it. If it's always namespace scoped, it should always come with the namespace. Adding extra check makes sense as well.